### ISSUE TITLE
"Otherwise" evaluated to early in switch node

### ISSUE THREAD

**Issue by overlinden** [node-red/node-red#4659]
### Current Behavior

Due to flow layout reasons I changed the order of the cases in the switch node to

1. otherwise
2. my condition

I recognized that the message is forwarded to both outputs which causes unexpected behavior.

### Expected Behavior

The order of the cases should be honored. Nevertheless I expect the "otherwise" to be evaluated at last wherever it is located in the conditions of the switch node.

### Steps To Reproduce

- Create a inject node that outputs 10.
- Add a switch node with two cases
- First case "otherwise"
- Second case "==10"
- Connect debug nodes to the outputs

You will see that the message "10" is printed for both outputs.

### Example flow

```
[
    {
        "id": "1b4565177e64597f",
        "type": "tab",
        "label": "Spielwiese",
        "disabled": false,
        "info": "",
        "env": []
    },
    {
        "id": "397f959470a9d42d",
        "type": "inject",
        "z": "1b4565177e64597f",
        "name": "",
        "props": [
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "10",
        "payloadType": "num",
        "x": 330,
        "y": 180,
        "wires": [
            [
                "d0b3cc0257b30930"
            ]
        ]
    },
    {
        "id": "d0b3cc0257b30930",
        "type": "switch",
        "z": "1b4565177e64597f",
        "name": "",
        "property": "payload",
        "propertyType": "msg",
        "rules": [
            {
                "t": "else"
            },
            {
                "t": "eq",
                "v": "10",
                "vt": "num"
            }
        ],
        "checkall": "true",
        "repair": false,
        "outputs": 2,
        "x": 530,
        "y": 180,
        "wires": [
            [
                "2bcffee607a97e7f"
            ],
            [
                "a364ae22bedf1205"
            ]
        ]
    },
    {
        "id": "a364ae22bedf1205",
        "type": "debug",
        "z": "1b4565177e64597f",
        "name": "debug 2",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 700,
        "y": 140,
        "wires": []
    },
    {
        "id": "2bcffee607a97e7f",
        "type": "debug",
        "z": "1b4565177e64597f",
        "name": "debug 3",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 700,
        "y": 220,
        "wires": []
    }
]
```

### Environment

I am running the official Node-Red docker image on a linux host.

**Comment by hardillb** [comment]
Otherwise is expected to be last

**Comment by overlinden** [comment]
Yes, this is the actual behavior and from a programmers perspective that is reasonable. After I found out that makes sense to me as well but this seems to be unintuitive for some non technical users.

**Comment by dceejay** [comment]
I don't think there is anything to fix here is there ?
OK to Close ?

**Comment by My-Random-Thoughts** [comment]
While this may be obvious to technical people, is it worth adding a note to the switch node to say rules are processed in order from top to bottom?

**Comment by GogoVega** [comment]
This might be a bit of a "stupid" message. This behavior is directly related to programming and not to Node-RED, I rather suggest a light learning in the `tour`(1) way (a few basic rules to get started). This idea can be very interesting for real novices. The question being is it profitable to code it?

[1] the window with the new features of the current version.

**Comment by mobamoba** [comment]
I'd just add one thing: the switch node defaults to "checking all rules" which, to my non-programming mind, certainly seems to say the order of the rules makes no difference. I note this because I, too, have sometimes put "otherwise" first only to discover my flow didn't work properly since I assumed the node was going to check them all since... that's what it said it would do.

If otherwise is always last from a programmer's POV, perhaps it could automatically be placed last when selected by the user or clarify what "checking all rules" means since that implies order is irrelevant.

**Comment by Steve-Mcl** [comment]
Being a programmer before ever touching Node-RED, I can say I have never put the `otherwise` option anywhere but at the end.

However, I can certainly see why a non programmer might reasonable expect `otherwise` at the top to work.

I am not actually advocating that we should reprogram the switch node to support this but rather we might consider adding some kind of visual feedback like a form-tip that is only shown when `otherwise` option is present and is NOT the last item?  Other options could be linting rules (though not everyone uses or installs the linter) or node validation (to show the orange triangle badge when `otherwise` is not the last item or there is more than 1 `otherwise` entry)?

**Comment by gdvine** [comment]
Typo in issue title: `too`

"otherwise" is *not* the same as an `else` block at the end of a condition block in code when it forms a draggable part of a user interface, and can't be assumed to be so whether the user is a dev or not.

This is poor UX design; either the functionality of `otherwise` should change to that expected, or `otherwise` should be hard-set in the UI to live only at the end of the condition block and not be draggable.

**Comment by overlinden** [comment]
@gdvine That is exactly my point as you pointd out in #5349 as well.
I wanted to put the `otherwise`  on top because I like having a clear layout in my node red flows. I want to avoid unnecessary crossing connections.

So, if there is a unwritten rule that the `otherwise` needs to be the last statement, the UI should avoid putting it somewhere else.

**Comment by gdvine** [comment]
> I wanted to put the `otherwise` on top because I like having a clear layout in my node red flows. I want to avoid unnecessary crossing connections.

This is exactly the use case.

This bug is a classic case of conflating visual design with logical infrastructure causing a UX contradiction and usability discrepancy. They must always be considered discretely for such features.
